apps/prairielearn/src/components/QuestionContainer.html.ts

```diff
@@ -401,14 +403,15 @@ function QuestionFooterContent({
                 </button>
               `
             : ''}
-          ${assessment?.group_config?.has_roles &&
-          !instance_question?.group_role_permissions?.can_submit
+          ${group_config?.has_roles && !instance_question?.group_role_permissions?.can_submit
```

Comment:
`instance_questions.group_role_permissions` is not set in the `renderPanelsForSubmission` context, so this will not be properly handled in case of external grader results.

Reply:
Hmm, good call. What do you think is the right way to handle this, just assign to `instance_question.group_role_permissions` there? This should get much more straightforward in #10792 when we can just rely on this being there from middleware.

Reply:
I think you should do the same thing you did for group_config and group_info, and save it directly to res.locals in the middleware, then add it to the panel code in renderPanelsForSubmission.

Reply:
Good point. I think I'd been hesitant to do that because on the assessment instance page, it _must_ be attached to individual instance questions since there are more than one of them. But that's only in one place, and TypeScript is already covering that (or at least it is as of this PR). Done in 4c1c5bd.